### PR TITLE
The file contains codes that convert MF4 to CSV file

### PR DESCRIPTION
**Pull request by oluyemiamujo**
The file contains codes that convert MF4 to CSV file, and each folder contains its own files.

### CODE REVIEW

**Comment by oluyemiamujo** [comment]
.

File: mf4_csv/README.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+Use the link to access the files
+
+https://drive.google.com/drive/folders/1TA09-u2tKYYJvLhUSmdaMHO4MHXSjwz_
```
Comment (ActuallyTaylor) [inline mf4_csv/README.md:4]: What does this file link out too?

File: mf4_csv/main.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    # mdf = MDF(f_in) #read mf4 data 
+
+    # d_dir = prepare_dir(mf4_file)
+    # print(d_dir)
+    f_out = mf4_file.replace(".MF4", ".csv")
+    data.export(fmt='csv', filename=f_out) #write to csv 
+    print("{} group completed".format(f_out))
+
+
+for root, dirs, files in os.walk("./"):
```
Comment (ActuallyTaylor) [inline mf4_csv/main.py:28]: Can this be moved into a python main function with a main guard? In the future this script could be included in something more high level that does batch work and being able to ensure the correct operation of the file will be nice.

This is some general information on python main guard https://stackoverflow.com/a/419185

File: mf4_csv/main.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    for filename in files:
+        # doSomethingWithFile(os.path.join(root, filename))
```
Comment (ActuallyTaylor) [inline mf4_csv/main.py]: Can these comments be removed?

File: mf4_csv/main.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        # if glob.glob(os.path.join(root, filename)+"/*.MF4"):        
+        ext = ["MF4", "mf4"];
+        if filename.endswith(ext[0], filename.index("."), len(filename)) or filename.endswith(ext[1], filename.index("."), len(filename)):
+            # print(os.path.join(root, filename))
+            convertMF4toCSV(os.path.join(root, filename))
+
+        
+print("All file converted!")
+    # for dirname in dirs:
```
Comment (ActuallyTaylor) [inline mf4_csv/main.py]: Same with these

File: mf4_csv/main.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import re
+import glob
+
+
+output_dir = "./output_in_csv"
+
+def prepare_dir(mf4_file):
+    my_str = str(mf4_file)
+    my_str = my_str.lower().strip(".mf4")
+    my_str = my_str.replace(".MF4", ".csv")
```
Comment (ActuallyTaylor) [inline mf4_csv/main.py:15]: Will this code actually replace the .MF4 extension? If it is being stripped in the previous statement with `my_str.lower().strip("mf4")` the string should no longer contain any file extension.

File: mf4_csv/main.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+from os import walk
+import re
+import glob
+
+
+output_dir = "./output_in_csv"
+
+def prepare_dir(mf4_file):
+    my_str = str(mf4_file)
+    my_str = my_str.lower().strip(".mf4")
```
Comment (ActuallyTaylor) [inline mf4_csv/main.py:14]: What would happen if my filename had some uppercase letters? Will this code still function?

File: mf4_csv/main.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    return my_str
+
+   
+
+def convertMF4toCSV(mf4_file):
+    data = MDF(mf4_file)
+    # f_out = f_in.replace(".MF4", ".csv") 
+    # mdf = MDF(f_in) #read mf4 data 
+
+    # d_dir = prepare_dir(mf4_file)
```
Comment (ActuallyTaylor) [inline mf4_csv/main.py]: Why is the directory preparation commented out?

**Review by ActuallyTaylor — CHANGES_REQUESTED** [review]

File: mf4_csv/README.md (hunk 1 above)
Comment (oluyemiamujo) [reply mf4_csv/README.md:4]: The link points to the shared folder that contains the MF4 and its corresponding CSV files. The size is big nd cannot be pushed to github

File: mf4_csv/main.py (hunk 3 above)
Comment (oluyemiamujo) [reply mf4_csv/main.py]: done

File: mf4_csv/main.py (hunk 4 above)
Comment (oluyemiamujo) [reply mf4_csv/main.py]: done

File: mf4_csv/main.py (hunk 7 above)
Comment (oluyemiamujo) [reply mf4_csv/main.py]: added

File: mf4_csv/main.py (hunk 2 above)
Comment (oluyemiamujo) [reply mf4_csv/main.py:28]: I will look into that tomorrow. I have HW to submit now.

File: mf4_csv/main.py (hunk 5 above)
Comment (oluyemiamujo) [reply mf4_csv/main.py:15]: The approach is to create a csv file with the same name as the original mf4 file in the same directory. The strip is to remove the extension and use the other part of the name and directory.

File: mf4_csv/main.py (hunk 6 above)
Comment (oluyemiamujo) [reply mf4_csv/main.py:14]: Sure! although that ll be tested asap.

File: mf4_csv/README.md (hunk 1 above)
Comment (ActuallyTaylor) [reply mf4_csv/README.md:4]: Sounds good

**Pull request closed without merging**